File: LeetCode/1995. Count Special Quadruplets/solution.py
"""
211 / 211 test cases passed.
Runtime: 1156 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def countQuadruplets(self, nums: List[int]) -> int:
        n, ans = len(nums), 0
        for a in range(n - 3):
            for b in range(a + 1, n - 2):
                for c in range(b + 1, n - 1):
                    for d in range(c + 1, n):
                        if nums[a] + nums[b] + nums[c] == nums[d]:
                            ans += 1
        return ans
    

File: LeetCode/1995. Count Special Quadruplets/solution.cpp
/**
 * 211 / 211 test cases passed.
 * Runtime: 132 ms
 * Memory Usage: 10.3 MB 
 */
class Solution {
public:
    int countQuadruplets(vector<int>& nums) {
        int n = nums.size(), ans = 0;
        for (int a = 0; a < n - 3; ++ a) {
            for (int b = a + 1; b < n - 2; ++ b) {
                for (int c = b + 1; c < n - 1; ++ c) {
                    for (int d = c + 1; d < n; ++ d) {
                        if (nums[a] + nums[b] + nums[c] == nums[d]) {
                            ++ ans;
                        }
                    }
                }
            }
        }
        return ans;
    }
};
